### PR TITLE
[feat] rewrite Cache Store to be async

### PR DESCRIPTION
**Pull request by kaushalkapasi**
- update DVC Client to properly use async Cache Store calls
- update tests to always `await client.onInitialized()` before interacting with the client

### CODE REVIEW

**Review by laurawarr — APPROVED** [review]
looks good to me

**Comment by laurawarr** [comment]
still looks good to me, but I'll defer to @rodeora and @L-Joli since they're more familiar with the changes we've made to the js sdk recently

File: sdk/js/src/CacheStore.ts [hunk 1]
```diff
@@ -17,13 +11,15 @@ export abstract class CacheStore {
         this.logger = logger
     }
 
-    private save(storeKey: string, data: unknown): void {
+    save(storeKey: string, data: unknown): void {
```
Comment (ajwootto) [inline sdk/js/src/CacheStore.ts:14]: separate save and load into their own `Storage` class and have this class take an instance of `Storage` in its constructor

File: sdk/js/src/CacheStore.ts [hunk 2]
```diff
@@ -38,46 +23,42 @@ export abstract class CacheStore {
...
 
-    private loadConfigFetchDate(user: DVCPopulatedUser): number {
+    private async loadConfigFetchDate(user: DVCPopulatedUser): Promise<number> {
         const fetchDateKey = this.getConfigFetchDateKey(user)
-        return parseInt(this.load<string>(fetchDateKey) || '0', 10)
-    }
-
-    remove(storeKey: string): void {
-        this.store?.removeItem(storeKey)
+        return parseInt(await this.store.load<string>(fetchDateKey) || '0', 10)
```
Comment (laurawarr) [inline sdk/js/src/CacheStore.ts:33]: splitting this into two lines could make it easier to read
```suggestion
        const fetchDate = await this.store.load<string>(fetchDateKey) || '0'
        return parseInt(fetchDate, 10)
```

**Review by laurawarr — APPROVED** [review]

**Pull request merged**